src/Features/Core/Portable/CodeFixes/ICodeFixService.cs

```diff
@@ -11,6 +12,8 @@ namespace Microsoft.CodeAnalysis.CodeFixes
 {
     internal interface ICodeFixService
     {
+        [Obsolete("This is a hack to support TypeScript; please do not use it.")]
+        Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document document, TextSpan textSpan, bool includeSuppressionFixes, bool isBlocking, CancellationToken cancellationToken);
```

Comment:
i'm confused.  we say it's a hack.  But all our roslyn code calls it intentionally (otherwise it wouldn't work).  It feels very necessary and feels like it *must* be used or else the TS scenario won't work.  

Souldn't we obsolete the previous method instead, so we catch any callers of it that we shoudl fixup?

Reply:
It's a hack in the sense that it deliberately violates several layers of abstraction.  However, I'm happy to remove the ObsoleteAttribute (other than on `IsBlocking`?) if you think that would be clearer.

Reply:
It doesn't feel like a hack to me at all.  FWIW, i think it would be good if *every* language supported this.  i.e. if you're in C#, it doesn't matter that C# is multithreaded.  The user is waiting for results, so they should be prioritized.

Actually... that may be a better way to go about this.  Rather than an IsBlocking flag, we have an 'IsHighPriority' flag.  It means the same thing, but it's effectively telling the end implementation: please make this happen right away since we need the results asap.

now there is no violation of layering.  The concept makes sense and is a core part of how the feature works and the contract we want from the impl.

@mavasani If we do this, can we also boost priority on the work that Roslyn is doing as well?  It doesn't make sense for us to be just twiddling our thumbs on this work when it is impacting the user.

Reply:
Sounds reasonable to me.